### ISSUE TITLE
execstats: tracking issue for top-level trace stats

### ISSUE THREAD

**Issue by asubiotto** [cockroachdb/cockroach#56239]
`execstats.TraceAnalyzer` is an object that can calculate top-level statement statistics from a physical plan and the execution trace of that plan.

This issue is a tracking issue for adding calculations of top-level stats from fine-grained execution stats collected from a trace. Note that this issue is only about the calculation of these stats. It is out of scope whether and how these stats will be displayed. That is still a question that will be resolved by the design team. Here is the current list of stats to calculate:

- [x] maximum flow memory usage (#54340) 
- [ ] cumulative time spent in operators (#56232) 
- [ ] cumulative time spent in network (#56576)
- [x] cumulative time spent waiting on KV (#56604)
- [x] total rows/bytes read from disk (#56605)
- [x] maximum flow temporary disk usage (#56606)
- [x] total network hops (#56607)

Jira issue: CRDB-2962
